chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -367,60 +377,130 @@ func (b *BitcoindNotifier) handleRelevantTx(tx chain.RelevantTx, bestHeight int3
 // historicalConfDetails looks up whether a transaction is already included in a
 // block in the active chain and, if so, returns details about the confirmation.
 func (b *BitcoindNotifier) historicalConfDetails(txid *chainhash.Hash,
+	heightHint, currentHeight uint32) (*chainntnfs.TxConfirmation, error) {
+
+	// First, we'll attempt to retrieve the transaction details using the
+	// backend node's transaction index.
+	txConf, err := b.confDetailsFromTxIndex(txid)
+	if err != nil {
+		return nil, err
+	}
+
+	if txConf != nil {
+		return txConf, nil
+	}
+
+	// If the backend node's transaction index is not enabled, then we'll
+	// fall back to manually scanning the chain's blocks, looking for the
+	// block where the transaction was included in.
+	return b.confDetailsManually(txid, heightHint, currentHeight)
+}
+
+// confDetailsFromTxIndex looks up whether a transaction is already included
+// in a block in the active chain by using the backend node's transaction index.
+// If the transaction is found, its confirmation details are returned.
+// Otherwise, a nil error is returned. A non-nil error may also be returned if
```

Comment:
Suggestion:
"If the transaction is found in the index, its confirmation details are returned. If it is not found in the index, a nil confirmation detail is returned." 
I think that should make it more clear that a `non-nil` error is _critical_(?).